drivers/include/at24cxxx.h

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_at24cxxx AT24CXXX EEPROM unit
+ * @brief       Device driver interface for the AT24CXXX EEPROM units
+ *
+ * @{
+ *
+ * @file
+ * @brief       Device driver interface for AT24CXXX EEPROM units.
+ *
+ * @author      Fabian Hüßler <fabian.huessler@ovgu.de>
+ *
+ */
+
+#ifndef AT24CXXX_H
+#define AT24CXXX_H
+
+#include <stdint.h>
+
+#include "periph/gpio.h"
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Return values
+ */
+enum {
+    AT24CXXX_OK,
+    AT24CXXX_I2C_ERROR
+};
+
+/**
+ * @brief   Struct that holds initialization parameters
+ */
+typedef struct at24cxxx_params {
+    i2c_t i2c;                      /**< I2C bus number */
+    gpio_t pin_wp;                  /**< write protect pin */
+    uint32_t eeprom_size;           /**< EEPROM memory capacity */
+    uint8_t dev_addr;               /**< I2C device address */
+    uint8_t page_size;              /**< page size */
+    uint8_t word_addr_len;          /**< data word address length */
```

Comment:
This member should not be needed. The `eeprom_size` should already be enough to determine if one or two address bytes are send. E.g.:

```C
uint8_t flags = (dev->eeprom_size > UINT8_MAX) ? I2C_REG16 : 0;
i2c_read_regs(.., flags, ...);
````

Reply:
Ah yes I can determine the number of address bits from the size...

Reply:
```C
uint8_t flags = (log2(dev->eeprom_size) > 11) ? I2C_REG16 : 0;
i2c_read_regs(.., flags, ...);
```

Reply:
Bad idea from my side. I overlooked that the multiplexing is also done for the devices with slightly more than 8 address bits.